crash entriesByBlockHashes

got crash with that text dump
```
2019-10-26 05:30:26.168 [INF] PEER: Growing size of checkpoint cache from 0 to 606 block hashes
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8305a3]

goroutine 181260 [running]:
github.com/gcash/bchd/blockchain/indexers.(*CfIndex).entriesByBlockHashes(0x0, 0x14cb330, 0x1, 0x1, 0x200, 0xc00bcd7900, 0x25e, 0x25e, 0x10, 0x10, ...)
        /home/travis/gopath/src/github.com/gcash/bchd/blockchain/indexers/cfindex.go:348 +0x153
github.com/gcash/bchd/blockchain/indexers.(*CfIndex).FilterHeadersByBlockHashes(...)
        /home/travis/gopath/src/github.com/gcash/bchd/blockchain/indexers/cfindex.go:386
main.(*serverPeer).OnGetCFCheckpt(0xc026e540d0, 0xc01e256000, 0xc026a3b380)
        /home/travis/gopath/src/github.com/gcash/bchd/server.go:1371 +0x5f7
github.com/gcash/bchd/peer.(*Peer).inHandler(0xc01e256000)
        /home/travis/gopath/src/github.com/gcash/bchd/peer/peer.go:1602 +0x12dc
created by github.com/gcash/bchd/peer.(*Peer).start
        /home/travis/gopath/src/github.com/gcash/bchd/peer/peer.go:2268 +0x29f
```

--- a/blockchain/chain.go
+++ b/blockchain/chain.go
@@ -2296,20 +2296,6 @@ func New(config *Config) (*BlockChain, error) {
 		return nil, err
 	}
 
-	// Initialize and catch up all of the currently active optional indexes
-	// as needed.
-	if config.IndexManager != nil {
-		err := config.IndexManager.Init(&b, config.Interrupt)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	// Initialize rule change threshold state caches.
-	if err := b.initThresholdCaches(); err != nil {
-		return nil, err
-	}
-
 	// If we're running in pruned mode or fast sync mode then set the chain type in the
 	// database. If not load the chain type from the database.
 	if config.Prune || config.FastSync {
@@ -2328,6 +2314,26 @@ func New(config *Config) (*BlockChain, error) {
 			}
 			return nil
 		})
+	}
+
+	// If we are pruned then done set any indexes as the manager cannot
+	// currently handle a pruned chain.
+	if b.isPruned {
+		b.indexManager = nil
+	}
+
+	// Initialize and catch up all of the currently active optional indexes
+	// as needed.
+	if b.indexManager != nil {
+		err := b.indexManager.Init(&b, config.Interrupt)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Initialize rule change threshold state caches.
+	if err := b.initThresholdCaches(); err != nil {
+		return nil, err
 	}
 
 	// Run an initial prune if prune mode is set

--- a/blockchain/indexers/manager.go
+++ b/blockchain/indexers/manager.go
@@ -271,11 +271,9 @@ func (m *Manager) Init(chain *blockchain.BlockChain, interrupt <-chan struct{}) 
 	}
 
 	// Migrate each index if necessary.
-	if !chain.IsPruned() {
-		for _, indexer := range m.enabledIndexes {
-			if err := indexer.Migrate(m.db, interrupt); err != nil {
-				return err
-			}
+	for _, indexer := range m.enabledIndexes {
+		if err := indexer.Migrate(m.db, interrupt); err != nil {
+			return err
 		}
 	}
 

